### ISSUE TITLE
Loss functions

### ISSUE THREAD

**Issue by GiorgioCerro** [GiorgioCerro/hyparticle-net#39]
Write the custom mAP loss function and combine it with the cross entropy

**Comment by GiorgioCerro** [comment]
The loss function has been implemented but it's extremely slow. Need to speed it up!
